### DISCUSSION TITLE
How to enable twig types for TypeScript/HTML Storybook?

### DISCUSSION THREAD

**Discussion by dibs** [larowlan/vite-plugin-twig-drupal#28] [Q&A] [answered]
I'm using TypeScript and can't get my Storybook to build due to missing twig types.
Is there a way to get this working in TS in a HTML Storybook?

Here's the error I get when I run $ npm run build
```
src/01-atoms/footer-logo/footer-logo.stories.ts:1:24 - error TS2307: Cannot find module './footer-logo.twig' or its corresponding type declarations.

1 import FooterLogo from "./footer-logo.twig"
```

**Comment by larowlan** [comment, marked as answer]
Drop this in .storybook/src/global.d.ts

```typescript

declare module "*.twig" {
  export default "" as (args) => string
}

```

Then make sure you're setup to read that in your tsconfig.json

**Reply by dibs** [reply]
Thanks Lee! I got my first dist folder.

For anyone else in the Twig in HTML with TS camp. 

I made the file above and then added the following to my tsconfig.json

```
"include": [
  "src",
  "global.d.ts"
]

```
Running npm run build threw errors that once worked through gave a build and then I could stand up Storybook again.
My life is complete.

**Comment by larowlan** [comment]
Full credit to @jptaranto for this solution
